### PR TITLE
Improve custom door support for custom models

### LINKED ISSUE #2643: Static door support for mods is lacking
**Describe the issue**

Static door (doors that transition between exteriors and interiors) support currently only works for model overrides of vanilla building models with DF door data. Each custom door references a static door on the original model for transitioning inside or back out again. The original doors can be suppressed from being active but the data must exist for a custom door to work. This use of existing static door data also restricts where the custom doors are located to one of the original doors for purposes of finding the closest start marker for positioning player inside.

This means there is currently no way to add doors to custom models, even if they're based on original models - if a custom model id is used, there is no door data available so it doesn't work.

Intention is to improve CustomDoor component. 

**Improvement todo list**

1. Allow custom building models to have transition doors defined that work.
2. Allow custom doors for custom dungeon entrance models.
3. Allow custom interior exit doors for custom interior models.

### PR DESCRIPTION
**Pull request by ajrb**
Improve custom door support for custom models, enabled for building exterior / interior and dungeon exterior doors. CustomDoor now creates its own StaticDoor data same as any DF model StaticDoor, allowing completely new custom models using non-classic model id's to be supported. 

Retained the existing copy mechanism that uses the boxcollider to test for a hit and the copied door data. This copy mechanism has been retained for backwards compatibility, code could be simplified if it were removed. I don't think it's that useful anymore. If any published mods use CustomDoor then they should continue to work identically as long as the copy index was set.

Player placement for both entering and exiting is now correct. The exception is when replacing a classic model with all building doors disabled in favour of custom doors all copying these classic doors.

Included 2 boxcollider presets for custom doors facing X or Z with standard DF door size 1.25, 2, 0.5. They're in Assets/Editor/ because Unity insisted they went in Assets folder and that seemed the most appropriate place. Will remove from PR if requested.

Ninelan will be testing this out to check my testing didn't miss anything. Instructions added to the wiki here: https://github.com/Interkarma/daggerfall-unity/wiki/Custom-Doors

Resolves #2643

### CODE REVIEW

File: Assets/Scripts/Utility/AssetInjection/Components/CustomDoor.cs
Comment (ajrb) [inline Assets/Scripts/Utility/AssetInjection/Components/CustomDoor.cs:96]: @KABoissonneault This part calculating the door normal may not be how it should be done, but it does seem to work. This was my 4th revision of this code and I am open to better methods if you want to suggest anything. My 3d graphics knowledge is rudimentary at best.

File: Assets/Scripts/Utility/RMBLayout.cs
Comment (ajrb) [inline Assets/Scripts/Utility/RMBLayout.cs:879]: The commented out line was what I thought I should use for the centre but the z is the same as parent transform position z which is incorrect. Unless anyone objects I will remove the commented out line.

Comment (ajrb) [reply Assets/Scripts/Utility/RMBLayout.cs]: Removed.

Comment (ajrb) [reply Assets/Scripts/Utility/RMBLayout.cs]: Actually it turns out calculating this works fine but only if the centre of the custom model is at the exact centre which is true for DF models but not true of the test custom model. Trying to find a solution to avoid all models having to have a dead centre 0,0,0

**Comment by ajrb** [comment]
So I've found a few issues when testing with Ninelan, am addressing them but hold off on reviewing this PR for now.

File: Assets/Scripts/Utility/RMBLayout.cs
Comment (ajrb) [reply Assets/Scripts/Utility/RMBLayout.cs]: Solved this now.

File: Assets/Scripts/Internal/DaggerfallInterior.cs
Comment (KABoissonneault) [inline Assets/Scripts/Internal/DaggerfallInterior.cs:485]: Do we have to call `GameObjectHelper.GetStaticDoors(ref modelData, entryDoor.blockIndex, entryDoor.recordIndex, modelMatrix)` a second time here, or would reusing `staticDoors` work?

**Comment by KABoissonneault** [comment]
For the most part, I'd like to unblock this and let ajrb figure out the feature. I just have one minor issue. Then, it should be ready to merge

File: Assets/Scripts/Internal/DaggerfallInterior.cs
Comment (ajrb) [reply Assets/Scripts/Internal/DaggerfallInterior.cs:485]: Yeah it would, I'll push a change when I get the chance to test it.

Comment (ajrb) [reply Assets/Scripts/Internal/DaggerfallInterior.cs]: Tested and committed.

**Review by KABoissonneault — APPROVED** [review]

**Pull request merged**
